gemrb/core/Streams/DataStream.cpp

```diff
@@ -135,20 +135,32 @@ strret_t DataStream::ReadLine(std::string& buf, strpos_t maxlen)
 	buf.clear();
 	buf.reserve(maxlen);
 	strpos_t i = 0;
-	//TODO: fix this to handle any combination of \r and \n
-	//Windows: \r\n
-	//Old Mac: \r
-	//otherOS: \n
+
+	bool isCR = false;
 	while (Pos < size && i < (maxlen - 1)) {
```

Comment:
this whole loop is really hard to read now.

Reply:
My mind works in state machines. Unless we want to introduce a `Peak` or something, that's what I found for compressing that a little:
```cpp
	while (Pos < size && i < (maxlen - 1)) {
		char ch;
		i += Read(&ch, 1);

		if (ch == '\r') {
			i += Read(&ch, 1);

			if (ch == '\n') {
				break;
			} else {
				i -= 1;
				Seek(-1, GEM_CURRENT_POS);
				break;
			}
		} else if (ch == '\n') {
			break;
		}

		if (ch == '\t') {
			ch = ' ';
		}

		buf.push_back(ch);
	}
```

Reply:
`Peek` would be useful. It's just read followed by seek.

I'm also fine with that change. Thanks.